bears/go/GoVetBear.py

```diff
@@ -25,5 +28,96 @@ class GoVetBear:
+            Locks that are erroneously passed by value.
+        :param method_signature:
+            Non-standard signatures for methods with familiar names, including:
+            Format, GobEncode, GobDecode, MarshalJSON, MarshalXML, Peek,
+            ReadByte,ReadFrom, ReadRune, Scan, Seek, UnmarshalJSON, UnreadByte,
+            UnreadRune, WriteByte, WriteTo
+        :param nil_function_compare:
+            Comparisons between functions and nil.
+        :param printf_call:
+            Suspicious calls to functions in the ``Printf`` family.
+        :param range_loop_variable:
+            Incorrect uses of range loop variables in closures.
+        :param shadow_variable:
+            Variables that may have been unintentionally shadowed.
+        :param shift:
+            Shifts equal to or longer than the variable's length.
+        :param struct_tag_format:
+            Struct tags that do not follow the format understood by
+            ``reflect.StructTag.Get``. Well-known encoding struct tags
+            (json, xml) used with unexported fields.
+        :param unreachable_code:
+            Unreachable code.
+        :param unsafe_pointer:
+            Likely incorrect uses of ``unsafe.Pointer`` to convert integers to
+            pointers.
+        :param unused_result:
+            Calls to well-known functions and methods that return a value that
+            is discarded.
+        """
+        # Every check in Vet is true by default.
+        # Adding a flag set to True enables that check and disables all other
+        # checks.
+        # Adding a flag set to False disables that check and runs all
+        # other checks.
+        vet_options = {'assembly_go_mismatch': '-asmdecl=true',
+                       'useless_assignment': '-assign=true',
+                       'atomic': '-atomic=true',
+                       'boolean': '-bool=true',
+                       'bad_build_tag': '-buildtags=true',
+                       'invalid_cgo': '-cgocall=true',
+                       'unkeyed_composite_literal': '-composites=true',
+                       'copying_lock': '-copylocks=true',
+                       'method_signature': '-methods=true',
+                       'nil_function_compare': '-nilfunc=true',
+                       'printf_call': '-printf=true',
+                       'range_loop_variable': '-rangeloops=true',
+                       'shadow_variable': '-shadow=true',
+                       'shift': '-shift=true',
+                       'struct_tag_format': '-structtags=true',
+                       'unreachable_code': '-unreachable=true',
+                       'unsafe_pointer': '-unsafeptr=true',
+                       'unused_result': '-unusedresult=true'}
+        frame = inspect.currentframe()
+        options, _, _, value_dict = inspect.getargvalues(frame)
+        values = [value_dict[i] for i in options[3:]]
+        valid_options = [vet_options[i] for i in options[3:]]
```

Comment:
uh looks like nice magic :) But I would maybe do it simpler:
```
vet_options = {
    '-asmdecl': assembly_go_mismatch,
    ...}

# Just a prototype, a better name would be nicce.
def to_vet_bool(val):
    return 'true' if val else 'false'

# Not sure actually what's better now... compress or dict-comprehension.
options = ('{}={}'.format(argname, to_vet_bool(value)) for argname, value in vet_options.items())
```

Reply:
As I've already mentioned, Vet works like this:
- If a flag is set to `true`, check corresponding to it will be performed and all other checks will be disabled.
- If a flag is set to `false`, only that check will be disabled and all other checks will be performed.

And because of https://github.com/coala/coala-bears/pull/2257#issuecomment-360585568, the flags not provided by the user should not be explicitly set to `false`.

So, I'm going with:
```python
def to_vet_bool(val):
    return 'true' if val
```

Reply:
As I've already mentioned, Vet works like this:

-If a flag is set to true, check corresponding to it will be performed and all other checks will be disabled.
- If a flag is set to false, only that check will be disabled and all other checks will be performed.
And because of #2257 (comment), the flags not provided by the user should not be explicitly set to false.

So, ```options = ('{}={}'.format(argname, to_vet_bool(value)) for argname, value in vet_options.items())``` won't work. `compress()` will do the workin this case.